Node2Code_Engine/Query/Expression.cs

```diff
@@ -0,0 +1,191 @@
+                SyntaxKind.ComplexElementInitializerExpression,
+                SyntaxFactory.SeparatedList(new List<ExpressionSyntax> {
+                    SyntaxFactory.LiteralExpression(SyntaxKind.StringLiteralExpression, SyntaxFactory.Literal(x.Item1.Name)),
+                    x.Item2
+                })
+            ).WithLeadingTrivia(SyntaxFactory.CarriageReturnLineFeed, SyntaxFactory.Tab) as ExpressionSyntax).ToList();
+
+            TypeSyntax dicType = SyntaxFactory.ParseTypeName(typeof(Dictionary<string, object>).ToText(true));
+            InitializerExpressionSyntax dicInitialiser = SyntaxFactory.InitializerExpression(SyntaxKind.ObjectInitializerExpression, SyntaxFactory.SeparatedList(dicAssignments));
+            ObjectCreationExpressionSyntax dicCreator = SyntaxFactory.ObjectCreationExpression(dicType, null, dicInitialiser);
+
+            List<ExpressionSyntax> assignments = new List<ExpressionSyntax>
+            {
+                 SyntaxFactory.AssignmentExpression(
+                     SyntaxKind.SimpleAssignmentExpression,
+                     SyntaxFactory.IdentifierName("CustomData"),
+                     dicCreator
+                 )
+            };
+
+            TypeSyntax objectType = SyntaxFactory.ParseTypeName(typeof(CustomObject).FullName);
+            InitializerExpressionSyntax initialiser = SyntaxFactory.InitializerExpression(SyntaxKind.ObjectInitializerExpression, SyntaxFactory.SeparatedList(assignments));
+            return SyntaxFactory.ObjectCreationExpression(objectType, null, initialiser);
+        }
+
+        /***************************************************/
+
+        public static ExpressionSyntax Expression(this LibraryNode node, List<ExpressionSyntax> inputs)
+        {
+            string dataName = "";
+            if (node.Outputs.Count > 0)
+                dataName = node.Outputs.First().Name;
+
+            ExpressionSyntax sourceExpression = SyntaxFactory.LiteralExpression(SyntaxKind.StringLiteralExpression, SyntaxFactory.Literal(node.SourceFile));
+            ExpressionSyntax dataExpression = SyntaxFactory.LiteralExpression(SyntaxKind.StringLiteralExpression, SyntaxFactory.Literal(dataName));
+
+            List<ArgumentSyntax> arguments = new ExpressionSyntax[] { sourceExpression, dataExpression }.Select(x => SyntaxFactory.Argument(x)).ToList();
+            ArgumentListSyntax argumentList = SyntaxFactory.ArgumentList(SyntaxFactory.SeparatedList(arguments));
+
+            return SyntaxFactory.InvocationExpression(SyntaxFactory.IdentifierName("BH.Engine.Library.Query.Match"), argumentList);
+        }
+
+
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        public static ExpressionSyntax Expression(this INode node, List<ExpressionSyntax> inputs)
```

Comment:
If this, and the methods below, are meant to be public, then - Documentation? If meant to be private then ignore